### PR TITLE
feat: age and entity groupings for datashield

### PR DESCRIPTION
**Pull request by jasminziegler**
for our federated analysis with DataSHIELD, we found that it is easier to have more subgroupings in our data e.g. to preprocess it even more so that we only have to do the bare minimum in the very restricted DataSHIELD environment.

### CODE REVIEW

**Review by jasminziegler — COMMENTED** [review]
self review

File: src/obds_fhir_to_opal/obds_fhir_to_opal.py [hunk 1]
```diff
@@ -144,7 +145,75 @@ def calculate_age_at_conditiondate(birthdate, conditiondate):
     return age_at_conditiondate
 
 
-def add_age_at_condition(df_pat_cond_joined):
+# 0-14, 15-19, 20-24, 25-29, 30-34, 35-39, 40-44, 45-49, 50-54, 55-59...80-84, 85+
+def diagnosis_age_group_small(age_at_diagnosis):
```
Comment (chgl) [inline src/obds_fhir_to_opal/obds_fhir_to_opal.py:149]: weiß nicht obs wirklich lesbarer ist, aber vielleicht:

```python
def diagnosis_age_group_small(age_at_diagnosis):
    age_ranges = [
        14,
        19,
        24,
        29,
        34,
        39,
        44,
        49,
        54,
        59,
        64,
        69,
        74,
        79,
        84,
        math.inf,
    ]

    for index, upper_limit in enumerate(age_ranges):
        if age_at_diagnosis <= upper_limit:
            return index
```

Comment (jasminziegler) [reply src/obds_fhir_to_opal/obds_fhir_to_opal.py:149]: macht sehr viel Sinn :D

File: src/obds_fhir_to_opal/obds_fhir_to_opal.py [hunk 2]
```diff
@@ -205,10 +290,89 @@ def lookup_condcodingcode_mapping(icd10_code):
     return icd10_code
 
 
+def group_entities(condcodingcode_mapped):
```
Comment (chgl) [inline src/obds_fhir_to_opal/obds_fhir_to_opal.py:293]: könnte auch sowas sein:

```py
def group_entities(condcodingcode_mapped):
    condcodingcode_mapped = float(condcodingcode_mapped)
    ranges = [
        (300, 315, 0),
        (315, 316, 1),
        (316, 317, 2),
        (318, 322, 3),
        (322, 323, 4),
        (323, 325, 5),
        (325, 326, 6),
        (332, 333, 7),
        (333, 335, 8),
        (343, 344, 9),
        (350, 351, 10),
        (353, 354, 11),
        (354, 356, 12),
        (356, 357, 13),
        (361, 362, 14),
        (362, 363, 15),
        (364, 365, 16),
        (367, 368, 17),
        (370, 373, 18),
        (373, 374, 19),
        (381, 382, 20),
        (382, 389, 21),
        (390, 391, 22),
        (391, 396, 23),
    ]

    for start, end, group in ranges:
        if start <= condcodingcode_mapped < end:
            return group

    return 100
```

**Pull request merged**